### ISSUE TITLE
cannot run `scripts/dev/setup`

### ISSUE THREAD

**Issue by despairblue** [facebook/nuclide#200]
```
Running pre-development setup scripts...
Nothing needed to be run.
Pre-development setup scripts executed.
START INSTALL: 2015-09-18 02:02:54.069339
Installing dependencies for Node package hyperclick-interfaces...
Process Process-1:
Traceback (most recent call last):
Installing dependencies for Node package nuclide-external-interfaces...
Process Process-2:
Traceback (most recent call last):
  File "/usr/lib/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
Installing dependencies for Node package nuclide-debugger-interfaces...
  File "scripts/dev/../lib/topological_installer.py", line 154, in install_dependencies
    package_json_sum = hashlib.sha1(read_file(package_json_path)).hexdigest()
TypeError: Unicode-objects must be encoded before hashing
Process Process-3:
Traceback (most recent call last):
  File "/usr/lib/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "scripts/dev/../lib/topological_installer.py", line 154, in install_dependencies
    package_json_sum = hashlib.sha1(read_file(package_json_path)).hexdigest()
TypeError: Unicode-objects must be encoded before hashing
  File "/usr/lib/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
Traceback (most recent call last):
  File "scripts/dev/setup", line 50, in <module>
    exclude_windows_incompatible_packages=is_windows())
  File "scripts/dev/../lib/package_manager.py", line 34, in install_dependencies
    installer.install()
  File "scripts/dev/../lib/topological_installer.py", line 53, in install
    self._do_multiprocess_install()
  File "scripts/dev/../lib/topological_installer.py", line 100, in _do_multiprocess_install
    (config['name'], exitcode))
Exception: Installing package nuclide-debugger-interfaces failed with exit code 1
  File "scripts/dev/../lib/topological_installer.py", line 154, in install_dependencies
    package_json_sum = hashlib.sha1(read_file(package_json_path)).hexdigest()
Installing dependencies for Node package nuclide-node-transpiler...
TypeError: Unicode-objects must be encoded before hashing
Process Process-4:
Traceback (most recent call last):
  File "/usr/lib/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "scripts/dev/../lib/topological_installer.py", line 154, in install_dependencies
    package_json_sum = hashlib.sha1(read_file(package_json_path)).hexdigest()
TypeError: Unicode-objects must be encoded before hashing
```

Could it be that it's incompatible with python 3?

**Comment by despairblue** [comment]
This is on arch linux.

**Comment by bolinfest** [comment]
The `hashlib.sha1()` code was a recent addition: we'll have to make something up that is compliant with Python 2 and 3. (Most of us use Python 2.7 by default since we're on OS X.)

In the meantime, you could probably comment that chunk of code out (or fix it!): it exists to avoid unnecessary calls to `npm install`, so it's not required for correctness, but it does speed things up dramatically.

**Comment by despairblue** [comment]
How about making it python 2 only and having a bash script that evaluates an environment variable that points to the correct python?
A little like npm does it: `env PYTHON=python2 npm i --save node-debugger`.

**Comment by ssorallen** [comment]
Looks like `read_file` needs to open the file in utf-8 encoding: http://stackoverflow.com/a/7585378/368697 I can fix that.

**Comment by nmote** [comment]
Fixed by 132c97d1dd1c07e45a9a0273deba3c6ed8f56796
